### PR TITLE
Added servicemonitor and fixed small issue in kubeclarity.labels help…

### PR DESCRIPTION
**Pull request by jmueller42**
## Description

This PR solves issue: https://github.com/openclarity/kubeclarity/issues/448

## Type of Change

[x] New Feature

## Checklist

- [X] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass

### CODE REVIEW

File: charts/kubeclarity/templates/clusterrole.yaml [hunk 1]
```diff
@@ -2,8 +2,7 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
   name: {{ include "kubeclarity.name" . }}
-  labels:
-    {{ include "kubeclarity.labels" . }}
+  labels: {{ include "kubeclarity.labels" . | nindent 4 }}
```
Comment (FrimIdan) [inline charts/kubeclarity/templates/clusterrole.yaml:5]: Hi @jmueller42 tnx for that.
I would prefer it will be put in place in a different PR and not along with the service monitor stuff.
Also, please, can you keep the format of
```
  labels:
    {{- include "kubeclarity.labels" . | nindent 4 }}
```

File: charts/kubeclarity/templates/service.yaml [hunk 2]
```diff
@@ -3,8 +3,8 @@ kind: Service
 metadata:
   name: {{ include "kubeclarity.name" . }}
   namespace: '{{ .Release.Namespace }}'
-  labels:
-    {{ include "kubeclarity.labels" . }}
+  labels: {{ include "kubeclarity.labels" . | nindent 4 }}
+    app: {{ include "kubeclarity.name" . }}
```
Comment (FrimIdan) [inline charts/kubeclarity/templates/service.yaml:7]: why we need `app: {{ include "kubeclarity.name" . }}` only on the service?
we can just add it to `kubeclarity.labels`, WDYT?

File: charts/kubeclarity/templates/service.yaml [hunk 3]
```diff
@@ -19,5 +19,11 @@ spec:
       port: {{ index .Values "kubeclarity-runtime-scan" "resultServicePort" }}
       protocol: TCP
       targetPort: {{ index .Values "kubeclarity-runtime-scan" "resultServicePort" }}
+    {{- if ((.Values.kubeclarity).prometheus).enabled }}
```
Comment (FrimIdan) [inline charts/kubeclarity/templates/service.yaml:22]: Please remove the unnecessary brackets
`{{- if .Values.kubeclarity.prometheus.enabled }}`

File: charts/kubeclarity/templates/servicemonitor.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{{- if (((.Values.kubeclarity).prometheus).serviceMonitor).enabled -}}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "kubeclarity.name" . }}
+  {{- if (((.Values.kubeclarity).prometheus).serviceMonitor).annotations }}
+  annotations: {{- toYaml (((.Values.kubeclarity).prometheus).serviceMonitor).annotations | nindent 4 }}
+  {{- end }}
```
Comment (FrimIdan) [inline charts/kubeclarity/templates/servicemonitor.yaml:9]: IMO better to use the `with` function
```
  {{- with .Values.kubeclarity.prometheus.serviceMonitor.annotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

File: charts/kubeclarity/templates/servicemonitor.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  labels: {{ include "kubeclarity.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels: {{- include "kubeclarity.labels" . | nindent 6 }}
+      app: {{ include "kubeclarity.name" . }}
+  endpoints:
+  - port: metrics
+    path: "/metrics"
+    interval: {{ (((.Values.kubeclarity).prometheus).serviceMonitor).interval }}
+{{- end -}}
```
Comment (FrimIdan) [inline charts/kubeclarity/templates/servicemonitor.yaml:19]: should be `{{- end }}`

File: charts/kubeclarity/templates/servicemonitor.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+{{- if (((.Values.kubeclarity).prometheus).serviceMonitor).enabled -}}
```
Comment (FrimIdan) [inline charts/kubeclarity/templates/servicemonitor.yaml:1]: Please remove the unnecessary brackets.
Also please remove the unnecessary separator `---`.
What about checking if `.Values.kubeclarity.prometheus.enabled`?

**Review by FrimIdan — CHANGES_REQUESTED** [review]

File: charts/kubeclarity/templates/clusterrole.yaml (hunk 1 above)
Comment (jmueller42) [reply charts/kubeclarity/templates/clusterrole.yaml:5]: okay

File: charts/kubeclarity/templates/service.yaml (hunk 2 above)
Comment (jmueller42) [reply charts/kubeclarity/templates/service.yaml:7]: good idea

File: charts/kubeclarity/templates/servicemonitor.yaml (hunk 6 above)
Comment (jmueller42) [reply charts/kubeclarity/templates/servicemonitor.yaml:1]: separator removed.

The brackets make it nil safe.

File: charts/kubeclarity/templates/service.yaml (hunk 3 above)
Comment (jmueller42) [reply charts/kubeclarity/templates/service.yaml:22]: The brackets make it nil safe. So i don't have to check 
```
{{ if .Values.kubeclarity }}
  {{ if .Values.kubeclarity.prometheus }}
    {{ if .Values.kubeclarity.prometheus.enabled }} 
...
```

File: charts/kubeclarity/templates/servicemonitor.yaml (hunk 5 above)
Comment (jmueller42) [reply charts/kubeclarity/templates/servicemonitor.yaml]: fixed

File: charts/kubeclarity/templates/servicemonitor.yaml (hunk 4 above)
Comment (jmueller42) [reply charts/kubeclarity/templates/servicemonitor.yaml]: Improved with your suggestion

File: charts/kubeclarity/templates/service.yaml (hunk 3 above)
Comment (FrimIdan) [reply charts/kubeclarity/templates/service.yaml:22]: I see, we don't do it in any other places, I don't want to start doing it here. we assume the path exists.

File: charts/kubeclarity/templates/clusterrole.yaml (hunk 1 above)
Comment (FrimIdan) [reply charts/kubeclarity/templates/clusterrole.yaml]: > I would prefer it will be put in place in a different PR and not along with the service monitor stuff.

@jmueller42  can you create 2 different PRs, 1 improving the `kubeclarity.labels` and the other one for the servicemonitor stuff?

Comment (jmueller42) [reply charts/kubeclarity/templates/clusterrole.yaml]: @FrimIdan The problem is the labels need to be indented 4 spaces in most of the templates but 6 spaces in the ServiceMonitor template

Comment (FrimIdan) [reply charts/kubeclarity/templates/clusterrole.yaml]: Not sure I understand how the indent is related.
I just asked to create 2 PRs
1. Improves the helper function: "kubeclarity.labels" in _helpers.tpl
2. Add servicemonitor

we will merge them one after the other.

Comment (jmueller42) [reply charts/kubeclarity/templates/clusterrole.yaml]: okay i will prepare 2 PRs

Comment (jmueller42) [reply charts/kubeclarity/templates/clusterrole.yaml]: @FrimIdan I created the first PR: https://github.com/openclarity/kubeclarity/pull/454

When this is merged i will update this one.

File: charts/kubeclarity/templates/service.yaml (hunk 3 above)
Comment (jmueller42) [reply charts/kubeclarity/templates/service.yaml:22]: i will change according to this

File: charts/kubeclarity/templates/clusterrole.yaml (hunk 1 above)
Comment (jmueller42) [reply charts/kubeclarity/templates/clusterrole.yaml]: @FrimIdan I updated this PR with the other changes

File: charts/kubeclarity/templates/servicemonitor.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.kubeclarity.prometheus.serviceMonitor.enabled -}}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "kubeclarity.name" . }}
+  {{- with (((.Values.kubeclarity).prometheus).serviceMonitor).annotations }}
```
Comment (FrimIdan) [inline charts/kubeclarity/templates/servicemonitor.yaml:6]: please remove the brackets

File: charts/kubeclarity/templates/servicemonitor.yaml [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    {{- include "kubeclarity.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "kubeclarity.labels" . | nindent 4 }}
```
Comment (FrimIdan) [inline charts/kubeclarity/templates/servicemonitor.yaml:15]: I think it should be `nindent 6`, right?

File: charts/kubeclarity/templates/servicemonitor.yaml (hunk 7 above)
Comment (jmueller42) [reply charts/kubeclarity/templates/servicemonitor.yaml]: fixed

File: charts/kubeclarity/templates/servicemonitor.yaml (hunk 8 above)
Comment (jmueller42) [reply charts/kubeclarity/templates/servicemonitor.yaml]: you are right. Fixed

**Review by FrimIdan — APPROVED** [review]

File: charts/kubeclarity/templates/servicemonitor.yaml [hunk 9]
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.kubeclarity.prometheus.serviceMonitor.enabled -}}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "kubeclarity.name" . }}
+  {{- with .Values.kubeclarity.prometheus.serviceMonitor.annotations }}
```
Comment (FrimIdan) [inline charts/kubeclarity/templates/servicemonitor.yaml:6]: you are missing a placeholder for `annotations` in `values.yaml`

Comment (jmueller42) [reply charts/kubeclarity/templates/servicemonitor.yaml:6]: fixed

**Review by FrimIdan — APPROVED** [review]
